### ISSUE TITLE
CCD100 IOC: reduce logging when disconnected

### ISSUE THREAD

**Issue by FreddieAkeroyd** [ISISComputingGroup/IBEX#2916]
As a supporter I would like the CCD100 IOC not to generate large logs on POLARIS when the device is not available/connected. Currently these are ~90MB per day.
 
The EPICS DB file currently has three independently 1 second scanning records reading values. A suggested mitigation would be to drop the scanning rate if errors are detected and then increase it again once errors are cleared (the SCAN field can be changed at run time)  by:

* Change to forward linked records driven by one scanning record rather than 3 independent 1 second scans
* Add calc record(s) to the processing chain  that look for alarms/errors, if all 3 reads failed then change scan time to 10 second, otherwise change/keep 1 second scan. Requiring all 3 to fail would stop the scan rate being changed on the odd transient read failure.

This should drops logs to 9MB/day while still allowing recovery without IOC restart when device is again available.

**Comment by DominicOram** [comment]
We should also work out why the IOC is actually getting any response at all. It looks like the issue is that the CCD100 has a high baud rate (57.6k) and there is a reasonably long cable between cabin and sample area. Potentially look at whether the CCD100 accepts a different baud or checksum.

**Comment by DominicOram** [comment]
The CCD100 can be run at a different baud rate but 57.6k is the factory default. As this is a very movable piece of sample environment kit, I'm not sure how feasible it will be to change the baud on all the devices around ISIS?

**Comment by DominicOram** [comment]
Main PR at https://github.com/ISISComputingGroup/EPICS-StreamDevice/pull/4

**Comment by FreddieAkeroyd** [comment]
Will review this tomorrow morning in time for sprint review

**Comment by FreddieAkeroyd** [comment]
@DominicOram Code looks OK. Do IOC tests work for you? They initially failed as they were looking for    READING:SP    rather than   SP    however when corrected they then fail with expected: 1.23 actual: 0.0

**Comment by DominicOram** [comment]
Do https://github.com/ISISComputingGroup/IBEX/issues/3142 at same time.

**Comment by Tom-Willemsen** [comment]
List of PRs to be reviewed:
- https://github.com/ISISComputingGroup/EPICS-DeviceEmulator/pull/52
- https://github.com/ISISComputingGroup/EPICS-StreamDevice/pull/4
- https://github.com/ISISComputingGroup/EPICS-IOC_Test_Framework/pull/63
- https://github.com/ISISComputingGroup/EPICS-ioc/pull/242
- https://github.com/ISISComputingGroup/EPICS-CCD100/pull/3
- https://github.com/ISISComputingGroup/EPICS-utilities/pull/8

**Comment by AdrianPotter** [comment]
@FreddieAkeroyd are you still reviewing this?

**Comment by FreddieAkeroyd** [comment]
@AdrianPotter yes, but I am still getting test failures - I will need to talk to @Tom-Willemsen
